tests/HttpRequests/create-donation.http

```diff
@@ -0,0 +1,4 @@
+POST http://localhost:30030/v1/donations
```

Comment:
What's this?

Reply:
> @bdsl I think broadly this approach is looking promising. My main concern is that it seems risky to lock in _all_ (or many) supported scenarios when we could benefit from a fix being out soon and don't really have time spare to load test the impact.

That's reasonable - do you think actually we may just need to get the lock from the stripe callback handler? It's a lock enforced by the DB not just an advisory lock we I suppose we only need to acquire it from one side of the possibly conflicting processes.

> (I can't comment on a 0 line file, but I'm also not clear what the empty `public/test.php` is doing?)

Sorry I must have committed that by mistake, I was running phpInfo from that file to work out how to use xdebug.

Reply:
This http file is something we can run in e.g. PhpStorm to manually test the create donation handler. Very minimal at the moment but may be useful. Or I can delete it if you think it will cause a problem.

Reply:
@bdsl I'm not totally sure to be honest. If going for just one side, my first instinct would be to lock in the donation API's `PUT` because we know that that's where the "bad" update can occur in terms of unexpectedly reverting the donation status – but it could be you're right that either is sufficient so long as both endpoints continue to use transactions that include the `find*(...)`.

It doesn't seem too likely to be that locking in both these places would increase the chance of failure for donors vs. locking in one, so perhaps we could try with the two for now and make a ticket to come back to assessing the minimum that works more rigorously after the campaign?

Reply:
Makes sense. I've reduced the amount we're using locking to just the update and stripe handler in https://github.com/thebiggive/matchbot/pull/453/commits/40f17efccd7bcc9632984e6d16238fba1cdb4917

Reply:
Nice, thanks @bdsl! LGTM